support/k8s/k8s-workload-registrar/webhookhandler.go

```diff
@@ -38,25 +117,76 @@ func (h *WebhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var in admv1beta1.AdmissionReview
-	if err := json.NewDecoder(req.Body).Decode(&in); err != nil {
-		http.Error(w, "Malformed JSON body", http.StatusBadRequest)
+	var body []byte
+	if data, err := ioutil.ReadAll(req.Body); err == nil {
+		body = data
+	}
+
+	obj, gvk, err := deserializer.Decode(body, nil, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Malformed JSON body: %v", err)
+		klog.Error(msg)
+		http.Error(w, msg, http.StatusBadRequest)
 		return
 	}
 
-	if in.Request == nil {
-		http.Error(w, "AdmissionReview is missing request", http.StatusBadRequest)
+	admit := newDelegateToV1AdmitHandler(h.controller.ReviewAdmission)
+	ctx := context.TODO()
+
+	var responseObj runtime.Object
+	switch *gvk {
+	case admv1beta1.SchemeGroupVersion.WithKind("AdmissionReview"):
+		requestedAdmissionReview, ok := obj.(*admv1beta1.AdmissionReview)
+		if !ok {
+			klog.Errorf("Expected v1beta1.AdmissionReview but got: %T", obj)
+			return
+		}
+		responseAdmissionReview := &admv1beta1.AdmissionReview{}
+		responseAdmissionReview.SetGroupVersionKind(*gvk)
+		resp, err := admit.v1beta1(ctx, *requestedAdmissionReview)
+		if err != nil {
+			msg := fmt.Sprintf("internal error occurred: %v", err)
+			klog.Error(msg)
+			http.Error(w, msg, http.StatusInternalServerError)
+			return
+		}
+		responseAdmissionReview.Response = resp
+		responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
+		responseObj = responseAdmissionReview
+	case admv1.SchemeGroupVersion.WithKind("AdmissionReview"):
+		requestedAdmissionReview, ok := obj.(*admv1.AdmissionReview)
+		if !ok {
+			klog.Errorf("Expected v1.AdmissionReview but got: %T", obj)
+			return
```

Comment:
Shouldn't we be calling an `http.Error` to fail the HTTP response? As it stands, this will cause the controller to return status 200.